src/js/utils/date-utils.js

```diff
@@ -0,0 +1,284 @@
+         * an unresolvable language or timezone
+         *
+         * preferred_language = 'en' yields Oct. 14, 2016 09:00 BST
+         *
+         * preferred_language = 'ru' yields 14 окт 2016 г. 09:00 BST
+         *
+         * @param {object} localTime A moment.js datetime object offset to a determined timezone.
+         * @param {string} language an ISO 639-1 language code (e.g. 'en-US', 'fr')
+         * @param {string} format (optional) A desired date format.
+         * @return {string} A language/timezone aware localized string
+        */
+        getDisplayTime = function(localTime, language, format) {
+            var displayTime = '';
+            var displayLang = _getDisplayLang(language);
+            var displayFormat = _getDisplayFormat(format);
+
+            if (_isValid(displayLang) && _isValid(displayFormat)) {
+                if (moment(localTime).isValid()) {
+                    displayTime = localTime.locale(displayLang).format(displayFormat);
+                    if (!_isValid(displayTime)) {
+                        displayTime = localTime.tz('UTC').locale('en').format(DATE_FORMATS.DEFAULT_DATE_TIME_FORMAT);
+                    }
+                } else {
+                    displayTime = '';
+                }
+            }
+            return displayTime;
+        };
+
+        /**
+         * Helper function to determine sys/browser language (if unpassed)
+         *
+         * @param {string} language (optional) an ISO 639-1 language code (e.g. 'en-US', 'fr')
+         * @returns {string} an ISO 639-1 language code (e.g. 'en-US', 'fr')
+         */
+        _getDisplayLang = function(language) {
+            var displayLang;
+            if (_isValid(language)) {
+                displayLang = language;
+            } else {
+                displayLang = window.navigator.userLanguage || window.navigator.language;
+            }
+
+            // default to en-US
+            if (_isValid(language)) {
+                return displayLang;
+            } else {
+                return 'en-US';
+            }
+        };
+
+        /**
+         * Helper function to determine format constant.
+         *
+         * Defaults to DEFAULT_DATE_TIME_FORMAT if unpassed
+         *
+         * @param {string} format (optional) A format constant.
+         * @returns {string} A format constant.
+         */
+        _getDisplayFormat = function(format) {
+            var displayFormat = DATE_FORMATS.DEFAULT_DATE_TIME_FORMAT;
+            if (_isValid(format) && format in DATE_FORMATS) {
+                displayFormat = DATE_FORMATS[format];
+            }
+            return displayFormat;
+        };
+
+        /**
+         * Helper function to determine variable/context validity.
+         *
+         * @param {string} candidateVariable to test
+         * @returns {boolean}
+         */
+        _isValid = function(candidateVariable) {
```

Comment:
The moment library will return 'Invalid date' if passed malformed variables. I've also managed to get 'None' from django/mako. Just trying to handle novices.